packages/ai-core/src/artifact-api.test.ts

```diff
@@ -0,0 +1,98 @@
+import nock from 'nock';
+import { HttpDestination } from '@sap-cloud-sdk/connectivity';
+import {
+  ArtifactApi,
+  ArtifactCreationResponse,
+  ArtifactList,
+  ArtifactPostData,
+  Artifact
+} from './index.js';
+
+describe('artifact', () => {
+  const destination: HttpDestination = {
+    url: 'https://ai.example.com'
+  };
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('get artifact parses a successful response', async () => {
+    nock(destination.url)
+      .get('/lm/artifacts')
+      .reply(
+        200,
+        {
+          count: 1,
+          resources: [
+            {
+              createdAt: '2024-07-08T13:36:41Z',
+              description: 'dataset for training test',
+              id: '08b9e4a4-2cea-4a3c-a509-9308d92bda85',
+              kind: 'dataset',
+              modifiedAt: '2024-07-08T13:36:41Z',
+              name: 'i745181-test-data',
```

Comment:
[suggestion] Please cleanup your personal id and also clean up actual ids from tests.
Valid for all tests.